### ISSUE TITLE
Client kick with the crash.

### ISSUE THREAD

**Issue by YOKOGAMA** [Funwayguy/BetterQuesting#399]
After removing the quest and placing a new one, client kick with the crash.
[crash-2018-06-11_00.57.48-client.txt](https://github.com/Funwayguy/BetterQuesting/files/2088271/crash-2018-06-11_00.57.48-client.txt)

**Comment by Vitali8** [comment]
The same issue, but it does with one of my administrators, not me.
[Crash](https://gist.github.com/Vitali8/8c79641d29a627b9efc20998ab8d619a)
And it happens when he tries to add a new quest to the grid: 
![image](https://user-images.githubusercontent.com/34043968/41823559-15cfd5e2-780a-11e8-9205-870f3bd37493.png)

**Comment by Funwayguy** [comment]
I'll admit, this took me way longer to figure out than it should have. Reason this happens is that when quests are deleted they sync the quests but not the quest lines client side. This means that when you add a new quest it uses that now 'free' ID but conflicts with the old entry not yet updated in the quest line. This also happens because I missed a secondary conflict check in quest lines (shouldn't be necessary if the prior code works but that's not the case here).

TLDR: I didn't delete quests properly and an additional missing check results in this crash.

I'm still trying to get a much bigger issue fixed elsewhere but I've fixed this in dev for the next update at least.

Side Note: Although this is probably the worst temporary solution, you can try running `/bq_user refresh` to possibly prevent the crash after deleting quests.
